OpenRA.Mods.Common/Traits/World/EditorResourceLayer.cs

```diff
@@ -113,6 +115,17 @@ public void UpdateCell(CPos cell)
 				Dirty.Add(cell + d);
 		}
 
+		void CalculateNetWorth()
```

Comment:
This is a bit ugly, and makes for a lot of unnecessary calculations.  You should instead update the state incrementally from inside UpdateCell.